Preload script fails to resolve trustedRenderer in dev mode

When switching projects in dev mode, the new `WebContentsView` fails to load its preload script, resulting in a blank screen with no IPC bridge available.

## Error

```
VM5 sandbox_bundle:2 Unable to load preload script: /Users/gpriday/Projects/Canopy/canopy-app/dist-electron/electron/preload.cjs
VM5 sandbox_bundle:2 Error: module not found: ../shared/utils/trustedRenderer.js
    at preloadRequire (VM5 sandbox_bundle:2:145068)
    at <anonymous>:9:30
    at runPreloadScript (VM5 sandbox_bundle:2:145337)
    at executeSandboxedPreloadScripts (VM5 sandbox_bundle:2:144610)
```

This cascades into CSP violations and React render errors since `window.electron` is never exposed:

```
index.html?projectId=...:898 Executing inline script violates the following Content Security Policy directive 'script-src 'self' 'wasm-unsafe-eval''
logger-Bqo_RAxI.js:1 [ERROR] [React] Uncaught render error Object
```

## Context

This happened during a project switch in dev mode, where the built output in `dist-electron/` can get stale relative to source. The preload script (`electron/preload.cts`) requires `../shared/utils/trustedRenderer.js` but the compiled output wasn't available at the resolved path.

This may only manifest when creating a new `WebContentsView` for a project that hasn't been opened yet in the current session — the initial window's preload loaded fine, suggesting the module resolution works on first boot but fails for subsequently created views under certain build states.

--- a/package.json
+++ b/package.json
@@ -45,7 +45,7 @@
     "lint:ci": "eslint . --max-warnings 0",
     "format": "prettier --write .",
     "format:check": "prettier --check .",
-    "typecheck": "tsc --noEmit && tsc -b electron/tsconfig.preload.json electron/tsconfig.json",
+    "typecheck": "tsc --noEmit && tsc -b electron/tsconfig.json && tsc --noEmit -p electron/tsconfig.preload.json",
     "check": "npm run typecheck && npm run lint:ratchet && npm run format:check",
     "fix": "npm run format && npm run lint:fix",
     "test:smoke": "node scripts/run-smoke.mjs",

